internetarchive/files.py

```diff
@@ -233,10 +236,8 @@ def download(self, file_path=None, verbose=None, ignore_existing=None,
                     return
 
         parent_dir = os.path.dirname(file_path)
-        if parent_dir != '' \
-                and not os.path.exists(parent_dir) \
-                and return_responses is not True:
-            os.makedirs(parent_dir)
+        if parent_dir != '' and return_responses is not True:
+            os.makedirs(parent_dir, exist_ok=True)
```

Comment:
This will throw an exception if a file by that intended directory name exists.

Reply:
I don't think that's a new issue.

The old code would have thrown an exception as well, if:
- Any of the directory components is a file, but the final path does not exist.
- If the parent directory exists as a file, it will probably throw here: https://github.com/jjjake/internetarchive/blob/master/internetarchive/files.py#L264

Reply:
I cannot follow your first point.

But it is true that the issue existed before. Still it now throws on the `os.makedirs` call where it before most likely threw on your marked line. I'm not sure what the ramifications of this are and whether that exception should be silenced to retain behaviour identical to the current program.

Reply:
If your parent dir is `a/b/c/d`, and `a/b` exists as a file, `os.path.exists("a/b/c/d") == False`, but `os.makedirs` will still error out due to not being able to make a directory under a file path - so "path component is a file during mkdir" was already an issue, the `exists()` check only ruled it out for the final component.

I also confirmed that you get `NotADirectoryError` for both `os.makedirs("a/b/c/d")`, and for `open("a/b/c/d", "w")`, so the exception line changes but it throws the same exception. IMO this is not a regression.

Reply:
I pushed a new version that catches OSError instead of FileNotFoundError for the os.remove, and moved this makedirs into the try block so it will be caught in the same way as previous open() error